File: sdks/typescript/src/resources/vectors.ts
import type { RequestClient } from "../core.js";
import type { DeleteResponse, UpsertResponse, VectorEntry } from "../types/index.js";

export class VectorsResource {
  constructor(private readonly _client: RequestClient) {}

  upsert(collection: string, vectors: VectorEntry[]): Promise<UpsertResponse> {
    return this._client._request(
      "POST",
      `/v1/collections/${encodeURIComponent(collection)}/vectors/upsert`,
      { json: { vectors } },
    );
  }

  delete(collection: string, ids: string[]): Promise<DeleteResponse> {
    return this._client._request(
      "POST",
      `/v1/collections/${encodeURIComponent(collection)}/vectors/delete`,
      { json: { ids } },
    );
  }
}

File: sdks/rust/src/resources/vectors.rs
use crate::client::BigRag;
use crate::core::urlencode;
use crate::error::BigRagError;
use crate::types::vectors::{DeleteResponse, UpsertResponse, VectorEntry};

/// Vectors resource — manage raw vectors directly.
pub struct Vectors<'a> {
    pub(crate) client: &'a BigRag,
}

impl Vectors<'_> {
    /// Upsert vectors into a collection.
    pub async fn upsert(
        &self,
        collection: &str,
        vectors: Vec<VectorEntry>,
    ) -> Result<UpsertResponse, BigRagError> {
        let path = format!("/v1/collections/{}/vectors/upsert", urlencode(collection));
        let body = serde_json::json!({ "vectors": vectors });
        self.client.transport.post(&path, &body).await
    }

    /// Delete vectors by ID from a collection.
    pub async fn delete(
        &self,
        collection: &str,
        ids: &[&str],
    ) -> Result<DeleteResponse, BigRagError> {
        let path = format!("/v1/collections/{}/vectors/delete", urlencode(collection));
        let body = serde_json::json!({ "ids": ids });
        self.client.transport.post(&path, &body).await
    }
}
